featuretools/io/sql.py

```diff
@@ -0,0 +1,100 @@
+import pandas as pd
+import sqlalchemy as sa
+
+import featuretools as ft
+
+
+def from_sql(id, connection=None, connection_string=None, tables=None, index={},
+             time_index={}, secondary_time_index={}, encoding={}, variable_types={}):
+    if connection is None:
+        if connection_string is None:
+            raise ValueError('connection or connection_string is required')
+        else:
+            connection = sa.create_engine(connection_string, echo=False).connect()
+
+    loader = EntitySetLoaderFromSQL(connection=connection, id=id, index=index, time_index=time_index,
+                                    secondary_time_index=secondary_time_index, encoding=encoding,
+                                    variable_types=variable_types)
+    loader.load_entityset(tables)
+    return loader.es
+
+
+def _find_pk_name(table):
+    column_names = table.columns.keys()
+    pks = [name for name in column_names if table.columns[name].primary_key]
+    if len(pks) > 1:
+        raise RuntimeError('Composite primary key detected')
+    elif len(pks) == 1:
+        return pks[0]
+    else:
+        return None
+
+
+class EntitySetLoaderFromSQL:
```

Comment:
this is an internal class, but brief documentation of each method would be helpful for maintainablility